pkg/ddc/alluxio/dataset.go

```diff
@@ -127,6 +127,11 @@ func (e *AlluxioEngine) UpdateDatasetStatus(phase datav1alpha1.DatasetPhase) (er
 			e.Log.Info("No need to update HCFS status")
 		}
 
+		datasetToUpdate.Status.FileNum, err = e.getDataSetFileNum()
```

Comment:
@TrafalgarZZZ WTDY? Is it fine that the fileNum is obtained in this logic? Or in the gorouting  that acquires the size of whole dataset.

Reply:
When syncMetadata(a goroutiune) done, the total count will be real size of whole dataset, i think it need`t a goroutiune.

Reply:
Since `TotalFileNums()` no longer has performance issue now, I don't think it needs a goroutine any more.

But in another way, I'm thinking given the fact that `UpdateDatasetStatus()` will be called once during setup of the dataset and many times during the dataset sync loop, if the function is called when metadata sync is actually not done (which means alluxio's metric `Master.TotalPath` is still increasing), `getDataSetFileNum()` would return a number that is going to increase every few seconds. Would it be too weird for a user who wants to observe the file number of the dataset?

Reply:
I think user more care about the rest time of loading data. Maybe it is an useful status, but not for user.

Reply:
I think the function `getDataSetFileNum` should be called after loading metadata is done.

Reply:
> I think user more care about the rest time of loading data. Maybe it is an useful status, but not for user.

Is it possible for the user to know the rest time of loading data, @apc999 ? I think it's really useful for user to know how long he needs to wait.

Reply:
@TrafalgarZZZ Is it better to get total file number in https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/alluxio/metadata.go#L15 ? I think it's the result of syncing metadata.

Reply:
DataLoad CRD also has a capability to load metadata from UFS, so if we extract total file num from the result of syncing metadata and make it constant(which means it won't be synced in the controller loop), how do we get the updated total file nums if someone else triggered load metadata by specified DataLoad CRD?